definitions/features/katello_service.rb

```diff
@@ -27,13 +29,67 @@ def make_start(spinner, options = {})
       spinner.update 'No katello service to start'
     else
       filters = "--only #{services.join(',')}"
-      spinner.update 'Starting the katello services..'
+      spinner.update 'Starting katello services..'
       execute!("katello-service start #{filters}")
     end
   end
 
+  def restart(spinner, options = {})
+    spinner.update 'Restarting katello services..'
+    if options[:only] || options[:exclude]
+      filters = construct_filters_for_restart(options)
+      execute!("katello-service restart #{filters}")
+    else
+      execute!('katello-service restart')
+    end
+  end
+
+  def run_with_hammer_ping_retry(spinner, action = '', services_options = {})
+    meth_call_by_action(action, spinner, services_options)
+    spinner.update 'Sometimes, services will take time to be ready'
+    begin
+      Timeout.timeout(30) do
```

Comment:
Is the Timeout necessary here? I believe the `hammer ping` will finish eventually: I would like to avoid us spawning more requests against the server until the the previous ones are processed.